### ISSUE TITLE
[autodev] Full sweep on PyTorch 2.12 RC (pre-release stable baseline)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#95]
## Goal

Run a full sweep on PyTorch 2.12 (RC / pre-release) to establish the new-stable baseline. PT 2.12 is being cut as the next stable release; we need a clean sweep against it before it ships.

## PT 2.12 wheel availability (verified 2026-04-29)

- **PyPI stable index** (`pip install torch`) — PT 2.12 NOT yet published. Latest stable is 2.11.0.
- **PyTorch test/RC index** (`pip install --index-url https://download.pytorch.org/whl/test/cu126 torch`) — PT 2.12.0+cu126 wheels available. Use this.
- Source build remains a fallback if RC channel proves unreliable.

## Setup

```bash
# Build the venv
python3 -m venv ~/envs/torch212-rc
source ~/envs/torch212-rc/bin/activate
pip install --index-url https://download.pytorch.org/whl/test/cu126 torch
pip install -r requirements.txt   # transformers, diffusers, etc.

# Verify
~/envs/torch212-rc/bin/python -c "import torch; print(torch.__version__)"  # expect 2.12.x
```

## Procedure

```bash
python3 tools/run_experiment.py sweep \
    --python ~/envs/torch212-rc/bin/python \
    --workers 4 \
    --output-dir sweep_results/baseline/pt2.12-rc/ \
    --source hf diffusers custom
```

This is a full sweep — both passes (identify + explain), eval + train modes, ~4-6h wall on A100.

## What to monitor

- Hygiene gate strict-mode count (today's classifier should keep this near zero with the new corpus + per-version known_errors filter; PT 2.12 has no version-scoped entries yet, so all bugs get re-verified)
- Compare against `sweep_results/baseline/pt2.11/` for regressions/improvements
- File new dynamo-team issues for any patterns that emerge (use `tools/file_issues.py` after sweep completes)

## Acceptance

- `sweep_results/baseline/pt2.12-rc/identify_results.json` exists
- `sweep_results/baseline/pt2.12-rc/explain_results.json` exists
- Compare report posted via `tools/compare.py sweep_results/baseline/pt2.11/identify_results.json sweep_results/baseline/pt2.12-rc/identify_results.json --labels 2.11 2.12-rc`
- Short summary posted to corpus chat (counts + headline regressions/wins)

## Glitches to watch for

- The RC wheel may have ABI quirks that source build doesn't hit (or vice versa). If RC sweep flakes, fall back to `scripts/build-nightly-from-source.sh` against the 2.12 release branch.
- `known_errors.json` has 0 entries scoped to 2.12 — every previously-known bug gets re-verified. Expect some surprises in the strict-mode bucket.

## Status

**Overnight AutoDev candidate.** Procedure is deterministic, bounded scope, no decisions to make mid-run. Output is verifiable in the morning. Same shape as #84 (correctness baseline) but for the regular sweep.

**Comment by penguinwu** [comment]
**Resolution: superseded by 2.12 nightly sweep + decision not to RC-sweep**

Per Peng (2026-05-01): "Don't re-sweep RC wheel — predicted delta too small to be worth ~3h compute."

What we did instead:
1. **Full sweep on PT 2.12 nightly** (`2.12.0.dev20260407+cu128`) completed 2026-04-30 — see `sweep_results/baseline/pt2.12-2026-04-30/`. Identify pass: 1604 work items, 1106 full_graph (70.0% on full set; 1019→1016 = -3 on the 1420-model intersection vs PT 2.11).
2. **Source-validated** that `pytorch/pytorch:main@HEAD` `init.py` is byte-identical to our 2.12 nightly's. Latest available 2.12 nightly (`2.12.0.dev20260408`) ships with the same Dynamo subsystem.
3. **PT 2.12 RC wheel** (`2.12.0+cu126` on `download.pytorch.org/whl/test/cu126/`, built ~2026-04-23 from `release/2.12` branch) inspected: only conservative cherry-picks (small Dynamo bug fixes, no big-feature PRs like #179611 deepcopy polyfill). Predicted RC sweep delta from our nightly: ≤10 status flips total — not worth the ~3h compute.

Detailed analysis in [PT 2.11 vs PT 2.12 comparison report (v2)](https://docs.google.com/document/d/1IgWSZnP--iOBGYUNazCI_JfkO_nA84u642OeAYqsauQ/edit). Filed Blt-regression upstream as pytorch/pytorch#182116 (the one real torch-side regression we found).

Closing this issue. Reopen if/when we want to validate against the actual 2.12.0 release wheel (when it ships) — but only if there's a specific concern; otherwise the nightly sweep is the right snapshot.
